core/src/main/java/com/datastrato/gravitino/catalog/SchemaOperationDispatcher.java

```diff
@@ -330,4 +303,103 @@ public boolean dropSchema(NameIdentifier ident, boolean cascade) throws NonEmpty
         ? droppedFromStore
         : droppedFromCatalog;
   }
+
+  private boolean importSchema(NameIdentifier identifier) {
+    EntityCombinedSchema combinedSchema = loadCombinedSchema(identifier);
+    if (combinedSchema.imported()) {
+      return false;
+    }
+
+    StringIdentifier stringId = getStringIdFromProperties(combinedSchema.schemaProperties());
+    long uid;
+    if (stringId != null) {
+      // If the entity in the store doesn't match the external system, we use the data
+      // of external system to correct it.
+      uid = stringId.id();
+    } else {
+      // If store doesn't exist entity, we sync the entity from the external system.
+      uid = idGenerator.nextId();
+    }
+
+    SchemaEntity schemaEntity =
+        SchemaEntity.builder()
+            .withId(uid)
+            .withName(identifier.name())
+            .withNamespace(identifier.namespace())
+            .withAuditInfo(
+                AuditInfo.builder()
+                    .withCreator(combinedSchema.auditInfo().creator())
+                    .withCreateTime(combinedSchema.auditInfo().createTime())
+                    .withLastModifier(combinedSchema.auditInfo().lastModifier())
+                    .withLastModifiedTime(combinedSchema.auditInfo().lastModifiedTime())
+                    .build())
+            .build();
+    try {
+      store.put(schemaEntity, true);
+    } catch (Exception e) {
+      LOG.error(FormattedErrorMessages.STORE_OP_FAILURE, "put", identifier, e);
+      throw new RuntimeException("Fail to access underlying storage");
+    }
+
+    return true;
+  }
+
+  private EntityCombinedSchema loadCombinedSchema(NameIdentifier ident) {
+    NameIdentifier catalogIdentifier = getCatalogIdentifier(ident);
+    Schema schema =
+        doWithCatalog(
+            catalogIdentifier,
+            c -> c.doWithSchemaOps(s -> s.loadSchema(ident)),
+            NoSuchSchemaException.class);
+
+    // If the Schema is maintained by the Gravitino's store, we don't have to load again.
+    boolean isManagedSchema = isManagedEntity(catalogIdentifier, Capability.Scope.SCHEMA);
+    if (isManagedSchema) {
+      return EntityCombinedSchema.of(schema)
+          .withHiddenPropertiesSet(
+              getHiddenPropertyNames(
+                  catalogIdentifier,
+                  HasPropertyMetadata::schemaPropertiesMetadata,
+                  schema.properties()))
+          .withImported(true);
+    }
+
+    StringIdentifier stringId = getStringIdFromProperties(schema.properties());
+    // Case 1: The schema is not created by Gravitino.
+    if (stringId == null) {
+      return EntityCombinedSchema.of(schema)
+          .withHiddenPropertiesSet(
+              getHiddenPropertyNames(
+                  catalogIdentifier,
+                  HasPropertyMetadata::schemaPropertiesMetadata,
+                  schema.properties()))
+          .withImported(isEntityExist(ident));
```

Comment:
Why do we need to check `isEntityExist`, what kind of scenario do you need to cover?

Reply:
First case: PG schema doesn't have properties. So we should judge it whether is imported by enitity exist instead of StringIdentifier.
Second case: If some table isn't created by us, they don't have stringIdentifier. But if we import it, the entity exist.

Reply:
You should comment on it, not explained here when we ask.

Reply:
OK, I will.

Reply:
Done.